lxd/db/instances.go

```diff
@@ -106,16 +106,15 @@ func (c *ClusterTx) GetNodeAddressOfInstance(ctx context.Context, project string
 		args = append(args, instType)
 	}
 
-	if strings.Contains(name, shared.SnapshotDelimiter) {
-		parts := strings.SplitN(name, shared.SnapshotDelimiter, 2)
-
+	instanceName, snapshotName, _ := strings.Cut(name, shared.SnapshotDelimiter)
```

Comment:
This has subtly changed behaviour?

Previously passing `foo/` as `name` would have caused the first if statement part to be run, using "" as the snapshot name.  Which I think would have then returned a not found error as no snapshot named "" exists.

But now using `foo/` will return an empty snapshotName, and the new logic will then use the else block with the instance name, which would find the "foo" instance potentially.

Instead, how about we use the third return value from strings.Cut as the condition for the if statement?